test/query_test/query_test.go

```diff
@@ -51,167 +49,308 @@ var _ = Describe("Service Manager Query", func() {
+						ResourceID:   serviceInstance2.ID,
+						ResourceType: web.ServiceInstancesURL,
+					}
+					latestOpForInstance2 := &types.Operation{
+						Base: types.Base{
+							ID: "latestOpForInstance2",
+						},
+						Type:         types.CREATE,
+						State:        types.SUCCEEDED,
+						ResourceID:   serviceInstance2.ID,
+						ResourceType: web.ServiceInstancesURL,
+					}
+
+					repository.Create(context.Background(), serviceInstance1)
+					repository.Create(context.Background(), serviceInstance2)
+					repository.Create(context.Background(), oldestOpForInstance1)
+					repository.Create(context.Background(), latestOpForInstance1)
+					repository.Create(context.Background(), oldestOpForInstance2)
+					repository.Create(context.Background(), latestOpForInstance2)
+				})
+
+				It("should return only the operations being last operation for their corresponding instances", func() {
+					criteria := []query.Criterion{query.ByIDExist(storage.GetSubQuery(storage.QueryForAllLastOperationsPerResource))}
+
+					list, err := repository.List(context.Background(), types.OperationType, criteria...)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(list.Len()).To(BeEquivalentTo(2))
+					Expect(listContains(list, "latestOpForInstance1"))
+					Expect(listContains(list, "latestOpForInstance2"))
+				})
+			})
 
-			id1 = createNotification(repository, now)
-			id2 = createNotification(repository, now.Add(-30*time.Minute))
 		})
+		Context("ById Exist with template parameters", func() {
+			When("There is an operation is associated to a resource and another operation that is resource-less", func() {
+				BeforeEach(func() {
+					common.RemoveAllInstances(ctx)
+					common.RemoveAllOperations(ctx.SMRepository)
+
+					resource := &types.Platform{
+						Base: types.Base{ID: "test-resource"},
+					}
+					opForInstance1 := &types.Operation{
+						Base: types.Base{
+							ID: "opForInstance1",
+						},
+						Type:         types.CREATE,
+						State:        types.SUCCEEDED,
+						ResourceID:   "test-resource",
+						ResourceType: web.ServiceInstancesURL,
+					}
+					resourcelessOperation := &types.Operation{
+						Base: types.Base{
+							ID: "resourcelessOp",
+						},
+						Type:         types.CREATE,
+						State:        types.SUCCEEDED,
+						ResourceID:   "NON_EXISTENT_RESOURCE",
+						ResourceType: web.ServiceInstancesURL,
+					}
+
+					_, err := repository.Create(context.Background(), resource)
+					_, err = repository.Create(context.Background(), opForInstance1)
+					Expect(err).ToNot(HaveOccurred())
+					repository.Create(context.Background(), resourcelessOperation)
+				})
+
+				It("should retrieve only the operation that is not associated to a resource", func() {
+					templateParameters := query.TemplateParameters{"RESOURCE_TABLE": "platforms"}
+					criterion := query.ByIDNotExistWithTemplateParameters(storage.GetSubQuery(storage.QueryForNonResourcelessOperations), templateParameters)
```

Comment:
aren't we in a context called "By ID Exist with template params"? this is testing ByIDNotExist